com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -3028,6 +3041,15 @@ public void Deserialize(FastBufferReader reader)
                 // The NetworkSceneHandle is the server-side relative
                 // scene handle that the NetworkObject resides in.
                 reader.ReadValue(out NetworkSceneHandle);
+
+                if (HasInstantiationPayload)
+                {
+                    if (NetworkManager.Singleton.PrefabHandler.TryGetPayloadSynchronizer(Hash, out INetworkInstantiationPayloadSynchronizer synchronizer))
+                    {
+                        var instantiationPayloadBufferReader = new BufferSerializer<BufferSerializerReader>(new BufferSerializerReader(reader));
```

Comment:
For my information, what is the requirement to instantiate a new buffer reader here, rather than using the existing reader?

Reply:
In this specific case, the original deserialization buffer isn't accessible within this method. To reuse the same buffer directly, I'd need to restructure the flow and move the read logic one layer up, likely into AddSceneObject.

However, that would split the serialization and deserialization paths across different areas of the codebase, making the overall logic harder to follow and maintain, similar to the complexity seen already with NetworkVariable handling.

Creating a new BufferSerializer instance here avoids that complication and keeps the deserialization step self-contained, easier to order and easier to reason about.

Reply:
I'll post another commit tomorrow to demonstrate what I mean. It might not be as bad as I initially thought, it could potentially follow the same pattern used by NetworkVariables, which might keep things cleaner than expected.

Reply:
Seems good to me, we can do that for sure, I will post a new commit in a few minutes with that change. That's an easy change.